e2e.go

```diff
@@ -21,6 +28,54 @@ type E2EConfig struct {
 	TargetNetworkPublicKey  string
 }
 
+const (
+	TX_SUCCESS = "SUCCESS"
+	TX_PENDING = "PENDING"
+)
+
+type RPCError struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+	Data    string `json:"data"`
+}
+
+type AccountInfo struct {
+	ID       string `json:"id"`
+	Sequence int64  `json:"sequence,string"`
+}
+
+type TransactionResponse struct {
+	ID     string `json:"id"`
+	Status string `json:"status"`
+}
+
+type RPCTransactionResponse struct {
+	Result TransactionResponse `json:"result"`
+	Error  *RPCError           `json:"error,omitempty"`
+}
+
+type TransactionStatusResponse struct {
+	ID            string `json:"id"`
+	Status        string `json:"status"`
+	EnvelopeXdr   string `json:"envelopeXdr,omitempty"`
+	ResultXdr     string `json:"resultXdr,omitempty"`
+	ResultMetaXdr string `json:"resultMetaXdr,omitempty"`
+}
+
+type RPCTransactionStatusResponse struct {
+	Result TransactionStatusResponse `json:"result"`
+	Error  *RPCError                 `json:"error,omitempty"`
+}
+
+type LedgerEntryResult struct {
+	XDR string `json:"xdr"`
+}
+
+type RPCLedgerEntryResponse struct {
+	Result LedgerEntryResult `json:"result"`
+	Error  *RPCError         `json:"error,omitempty"`
+}
+
```

Comment:
I think we should be using the soroban-rpc code for this so that it's reusable at other places. We may want to move the responses out of the `internal` directory and exponse them in a client library.

Reply:
> I think we should be using the soroban-rpc code for this so that it's reusable at other places. We may want to move the responses out of the `internal` directory and exponse it through a client library.

yes, an sdk lib/package for the rpc interface would be good to refer to from here, I didn't want the test referring to internals, I can create a ticket to capture that, and refactor this to that as part of it, does that sound good for doing in iterations?